modules/@apostrophecms/schema/lib/addFieldTypes.js

```diff
@@ -1116,49 +1110,58 @@ module.exports = (self) => {
-        .toArray();
-      // Must maintain input order. Also discard things not actually found in
-      // the db
-      const actualDocs = [];
-      for (const item of input) {
-        if ((typeof item) === 'string') {
-          const result = results
-            .find(result => (result.title === item) || (result._id === item));
-          if (result) {
-            actualDocs.push(result);
-          }
-        } else if ((item && (typeof item._id === 'string'))) {
-          const result = results.find(doc => (doc._id === item._id));
-          if (result) {
-            if (field.schema) {
-              const destArray = Array.isArray(destination[field.name])
-                ? destination[field.name]
-                : [];
-              const destItem = destArray.find((doc) => doc._id === item._id);
-              result._fields = {
-                ...destItem?._fields || {}
-              };
+      } else {
+        const results = await manager
+          .find(req, { $or: clauses })
+          .relationships(false)
+          .toArray();
+        // Must maintain input order. Also discard things not actually found in
+        // the db
+        const actualDocs = [];
+        for (const item of input) {
+          if ((typeof item) === 'string') {
+            const result = results
+              .find(result => (result.title === item) || (result._id === item));
+            if (result) {
+              actualDocs.push(result);
+            }
+          } else if ((item && (typeof item._id === 'string'))) {
+            const result = results.find(doc => (doc._id === item._id));
+            if (result) {
+              if (field.schema) {
+                const destArray = Array.isArray(destination[field.name])
+                  ? destination[field.name]
+                  : [];
+                const destItem = destArray.find((doc) => doc._id === item._id);
+                result._fields = {
+                  ...destItem?._fields || {}
+                };
 
-              if (item && ((typeof item._fields === 'object'))) {
-                await self.convert(
-                  req,
-                  field.schema,
-                  item._fields || {},
-                  result._fields,
-                  options
-                );
+                if (item && ((typeof item._fields === 'object'))) {
+                  await self.convert(
+                    req,
+                    field.schema,
+                    item._fields || {},
+                    result._fields,
+                    options
+                  );
+                }
               }
+              actualDocs.push(result);
             }
-            actualDocs.push(result);
           }
         }
+        destination[field.name] = actualDocs;
+      }
+      if (field.required && (destination[field.name].length === 0)) {
+        throw self.apos.error('required');
+      }
+      if (field.min && field.min > destination[field.name].length) {
```

Comment:
same checks as before, but applying them to the actual related documents found, not just the user's alleged input. (This was never a security hole, but it makes more sense this way.)